src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -46,54 +65,606 @@ def gt_change_transient_strides(
+
+    for state in sdfg.states():
+        for dnode in state.data_nodes():
+            if dnode.data != data_name:
+                continue
+            gt_propagate_strides_from_access_node(
+                sdfg=sdfg,
+                state=state,
+                outer_node=dnode,
+                processed_nsdfgs=processed_nsdfgs,
+                ignore_symbol_mapping=ignore_symbol_mapping,
+            )
+
+
+def gt_propagate_strides_from_access_node(
+    sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    outer_node: dace_nodes.AccessNode,
+    ignore_symbol_mapping: bool = True,
+    processed_nsdfgs: Optional[set[PropagatedStrideRecord]] = None,
+) -> None:
+    """Propagates the stride of `outer_node` to any adjacent NestedSDFG.
+
+    The function will propagate the strides of the data descriptor `outer_node`
+    refers to along all adjacent edges of `outer_node`. If one of these edges
+    leads to a NestedSDFG then the function will modify the strides of data
+    descriptor within to match the strides on the outside. The function will then
+    recursively process NestedSDFG.
+
+    It is important that this function will only handle the NestedSDFGs that are
+    reachable from `outer_node`. To fully propagate the strides the
+    `gt_propagate_strides_of()` should be used.
+
+    Args:
+        sdfg: The SDFG to process.
+        state: The state where the data node is used.
+        edge: The edge that reads from the data node, the nested SDFG is expected as the destination.
+        outer_node: The data node whose strides should be propagated.
+        ignore_symbol_mapping: If `False` (default is `True`), try to modify the `symbol_mapping`
+            of NestedSDFGs instead of manipulating the data descriptor.
+        processed_nsdfgs: Set of NestedSDFG that were already processed and will be ignored.
+            Only specify when you know what your are doing.
+    """
+    if processed_nsdfgs is None:
+        # For preventing the case that nested SDFGs are handled multiple time.
+        processed_nsdfgs = set()
+
+    for in_edge in state.in_edges(outer_node):
+        gt_map_strides_to_src_nested_sdfg(
+            sdfg=sdfg,
+            state=state,
+            edge=in_edge,
+            outer_node=outer_node,
+            processed_nsdfgs=processed_nsdfgs,
+            ignore_symbol_mapping=ignore_symbol_mapping,
+        )
+    for out_edge in state.out_edges(outer_node):
+        gt_map_strides_to_dst_nested_sdfg(
+            sdfg=sdfg,
+            state=state,
+            edge=out_edge,
+            outer_node=outer_node,
+            processed_nsdfgs=processed_nsdfgs,
+            ignore_symbol_mapping=ignore_symbol_mapping,
+        )
+
 
-def _find_toplevel_transients(
+def gt_map_strides_to_dst_nested_sdfg(
     sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    edge: dace.sdfg.graph.Edge,
+    outer_node: dace.nodes.AccessNode,
+    ignore_symbol_mapping: bool = False,
```

Comment:
Strange to see different default values for `ignore_symbol_mapping`. In any case, as commented above, I am of the opinion that this should not be configurable and symbol mapping should always be used.

Reply:
I also saw them, it was an error on my side.